### PR TITLE
Resource module for ios_l3_interfaces

### PR DESCRIPTION
**Pull request by justjais**
##### SUMMARY

Resource module for `ios_l3_interfaces`, with supported operations and respective integration tests.
(ref model: [ios_l3_interfaces](https://github.com/ansible-network/resource_module_models/pull/16) model)

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

ios_l3_interfaces
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/network/ios/ios_l3_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+
+DOCUMENTATION = """
+---
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ios/ios_l3_interfaces.py:40]: Add notes section with tested against version, the connection type to be used for this module and link to platform guide. Refer https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L106

**Review by ganeshrn — APPROVED** [review]
Apart from the notes comment, other changes looks good.

**Comment by justjais** [comment]
recheck

**Comment by ikhan2010** [comment]
recheck

**Comment by justjais** [comment]
recheck

**Review by danielmellado — COMMENTED** [review]
I'm seeing some loopback devices no the overridden test: ```[
    {
        "name": "loopback888"
    },
    {
        "name": "loopback999"
    },```

Are these expected to be there after override the config?  I don't think so and if that's the case maybe the test setup should enforce that these are deleted.

**Comment by justjais** [comment]
@danielmellado wrt to `loopback 888/999` in case of overridden too we're not going to delete the interface be it logical/physical. So, for before and after interface `loopback 888/999` will be present.

**Comment by justjais** [comment]
recheck

**Comment by justjais** [comment]
recheck

**Pull request merged**